### ISSUE TITLE
Jaws reads first checkbox from list as 'readonly' on datagrid Show/Hide columns dialog 

### ISSUE THREAD

**Issue by sumegha26** [vmware/clarity#4513]
## Describe the bug

Jaws reads first checkbox from list as 'readonly' on Show/Hide columns dialog.This is bug in Jaws only because NVDA reads it as 'list'.

## How to reproduce
- Issue can be easily reproducible on datagrid Show/Hide columns dialog here -
https://clarity.design/documentation/datagrid/hide-show

1.  Go to Hide/Show example on https://clarity.design/documentation/datagrid/hide-show
2.  Click on view-columns icon on footer
3.  Press enter to open the dialog
4.  Press tab two times to focus on first checkbox,focused element is `<input>` checkbox

Jaws reads `readonly` before first checkbox label , plus does not read as list. This is `<li>` under `<ul>` on checkboxes.

## Expected behavior

Jaws should not read 'readonly' on first checkbox focus

## Additional notes

 <input> checkbox has wrapper `<clr-checkbox-wrapper>` no role is specified on it , So it shows role 'generic' in accessibility tree.

**Workaround -**

1. Adding role on `<clr-checkbox-wrapper>` i.e. **role='none'**  Jaws stop announcing it as 'readonly'

even by  JFYI removing `<clr-checkbox-wrapper>` also  Jaws stop announcing it as 'readonly'

**Comment by gnomeontherun** [comment]
Clarity cares deeply about accessibility, and we work with our accessibility team to identify and triage issues. We have an internal tracking system to identify and verify accessibility issues, which has captured this already. Some issues with Clarity Angular are difficult to solve without causing side effects or breaking changes, and we have to balance concerns with fixing issues and breaking changes. As we build out Clarity Core, we are able to address these root cause accessibility issues and recommend adopting the Clarity Core implementations as they become available. We are closing this issue as we have the issues tracked internally. Please follow our development and releases to see when we release relevant Clarity Core components to make this possible.
